lib/std/crypto/tls/Client.zig

```diff
@@ -468,7 +468,7 @@ pub fn init(stream: anytype, ca_bundle: Certificate.Bundle, host: []const u8) In
                         read_seq += 1;
                         P.AEAD.decrypt(cleartext, ciphertext, auth_tag, record_header, nonce, p.server_handshake_key) catch
                             return error.TlsBadRecordMac;
-                        break :c cleartext;
+                        break :c @constCast(mem.trimRight(u8, cleartext, "\x00"));
```

Comment:
This looks unfortunate, is there an issue for `std.mem` changing constness?

Reply:
Agreed it looks bad, I had a discussion about this a while ago in the community discord but I don't believe there's an issue filed about it. 

It'd be nice if these functions changed their return type based on the input's constness.

Reply:
because its not ergonomic to do so and mutable slices coerce to const ones